### DISCUSSION TITLE
❌ 2FA (Authy) prompt with SFTP

### DISCUSSION THREAD

**Discussion by killianeletellier** [isontheline/pro.webssh.net#330] [Ideas]
Currently, 2FA prompt is only available with SSH. Without that feature with the SFTP tab, I can't connect to my SFTP server. 

So, I think it could be very useful to add this feature on the SFTP connections.

**Comment by isontheline** [comment]
Bonjour @LProgead 👋

Votre demande est relative à https://github.com/isontheline/pro.webssh.net/issues/174 et https://github.com/isontheline/pro.webssh.net/issues/51

Votre implémentation 2FA concerne-t-elle [DUO](https://duo.com/product/multi-factor-authentication-mfa) ?

**Reply by killianeletellier** [reply]
Bonjour,
Non, je n'utilise pas DUO, mais Authy.

**Reply by isontheline** [reply]
Auriez-vous un lien car je ne trouve que des ressources de 2018 ?

**Reply by killianeletellier** [reply]
Un lien à quel sujet ?
Du traitement de l'A2F pour SSH par Authy ?
Si oui, ils ont repo GitHub à [ce lien](https://github.com/authy/authy-ssh) qui, en effet, n'a pas obtenu de mise à jour depuis 2018.

**Reply by isontheline** [reply]
Oui c'était par rapport à la vétusté du dépôt effectivement.

Entendu je vais faire le nécessaire pour mettre en oeuvre le 2FA au sein du SFTP.

Bonne soirée à vous 🌓

**Reply by killianeletellier** [reply]
Merci beaucoup pour cette réactivité :D 

Bonne journée !

**Reply by isontheline** [reply]
@LProgead Pourriez-vous essayer la dernière beta de WebSSH ?
https://testflight.apple.com/join/QSrBK59z

**Reply by killianeletellier** [reply]
Bonjour @isontheline !
Je viens de l'essayer et... rien n'a changé. 
L'application charge toujours infiniment sans me demander aucun code d'A2F.

**Reply by isontheline** [reply]
Bonjour @LProgead 👋

Y compris avec l'option "Force Keyboard Interactive" active ?

![image](https://user-images.githubusercontent.com/44212923/125765144-4d498a44-556a-4702-a1bf-5400fcde260b.png)

**Reply by isontheline** [reply]
@LProgead Pardon on parle ici de SFTP et non de SSH... 😅

OK je regarde à nouveau

**Reply by isontheline** [reply]
@LProgead La création de compte Authy n'est plus possible. Les pages redirigent vers twilio.

Il vaudrait mieux upgrader vers Google Authenticator ou Duo.

**Reply by killianeletellier** [reply]
@isontheline, oui. C'est totalement normal. Authy appartient à Twilio. C'est à leur travers qu'il faut d'authentifier.

**Reply by isontheline** [reply]
@LProgead Je transfère la suite de mes investigations sur https://github.com/isontheline/pro.webssh.net/issues/346

N'hésitez pas à suivre cette issue pour toute information complémentaire.
